x/concentrated-liquidity/lp.go

```diff
@@ -37,16 +37,39 @@ func (k Keeper) createPosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddr
 		return sdk.Int{}, sdk.Int{}, sdk.Dec{}, err
 	}
 
-	// calculate liquidity created from this position
+	// N.B. we only write cache context if actual amounts
+	// returned are greater than the given minimums.
+	cacheCtx, writeCacheCtx := ctx.CacheContext()
+
+	currSqrtPrice := pool.GetCurrentSqrtPrice()
+	currentTick := pool.GetCurrentTick()
+
+	// If the currentSqrtPrice and currentTick are zero, then this is the first position to be created for this pool.
+	// We therefore calculate the sqrtPrice and currentTick based on the inputs of this position
+	// TODO: We need to review this logic very carefully, as I am not 100 percent convinced it is safe / correct
+	if currSqrtPrice.Equal(sdk.ZeroDec()) && currentTick.Equal(sdk.ZeroInt()) {
```

Comment:
Can we encapsulate all this into a method `isInitialPosition(...)` and cover by tests?